packages/controller-utils/src/util.ts

```diff
@@ -345,9 +353,30 @@ export function isValidHexAddress(
     return false;
   }
 
-  return isValidAddress(addressToCheck);
+  return isHexChecksumAddress(addressToCheck);
```

Comment:
Maybe this? Not sure, but at least that explains why `isValidHex` rely on the "checksum" variant function, WDYT?

```suggestion
  // We used to rely on `isValidAddress` from `@ethereumjs/util` which allows
  // for upper-case characters too. So we preserve this behavior and use our
  // faster and memoized validation function instead.
  return isHexChecksumAddress(addressToCheck);
```